libs/gl-client/.resources/proto/glclient/scheduler.proto

```diff
@@ -398,3 +425,134 @@ message SignerRejection {
+message PairDeviceRequest {
+	// The tls public key of the new device.
+	string session_id = 1;
+
+	// The certificate signing request that will be signed by the
+	// greenlight backend if the pairing succeeds. Notice that the CN
+	// here is irrelevant.
+	bytes csr = 2;
+
+	// The name of the device that will be part of the tls certificate
+	// subjects CN field: CN=/users/<node_id>/<device_name>.
+	string device_name = 3;
+
+	// A human readable description of the device, this can be a 
+	// purpose or something similar. Can be used to display to the 
+	// user on the old device.
+	string desc = 4;
+
+	// A set of restrictions that the new devices requests for the 
+	// rune. This might in the future get upgraded for easier naming.
+	string restrs = 5;
+}
+
+message PairDeviceResponse {
+	// Session id is the public key of the new device used for the
+	// tls cert.
+	string session_id = 1;
+
+	// Upon a pairing request, the device receives back the signed certificate
+	// that belongs to the certificate signing request the that was sent with
+	// the pairing request, so they can authenticate themselves in the future.
+	string device_cert = 2;
+
+	// The private key that was used to create the certificate with. This key
+	// is used to sign the requests to the node.
+	string device_key = 3;
+
+	// A rune that is returned if the device that is created by the signer that
+	// the device pairs to.
+	string rune = 4;
+
+	// Creds contains a serialized version of the device certificate, the device 
+	// key and the rune that are used to authenticate a device at the backend, 
+	// and to authorize a request at the signer.
+	bytes creds = 5;
+}
+
+message PairingQr {
+	// Holds the data that the new device shows the old device during a pairing 
+	// request. The data is of the form: gl-pairing:<session_id>
+	string data = 1;
+}
+
+message GetPairingDataRequest {
+	// The session_id that part of the qr code.
+	string session_id = 1;
+}
+
+message GetPairingDataResponse {
+	string session_id = 1;
+
+	// The certificate signing request that will be signed by the
+	// greenlight backend if the pairing succeeds. Notice that the CN
+	// here is irrelevant.
+	bytes csr = 2;
+
+	// The name of the device that will be part of the tls certificate
+	// subjects CN field: CN=/users/<node_id>/<device_name>.
+	string device_name = 3;
+
+	// A human readable description of the device, this can be a 
+	// purpose or something similar. Can be used to display to the 
+	// user on the old device.
+	string desc = 4;
+
+	// A set of restrictions that the new devices requests for the 
+	// rune. This might in the future get upgraded for easier naming.
+	string restrs = 5;
+}
+
+message ApprovePairingRequest {
+	string session_id = 1;
+
+	// The time that the old device approved the pairing request. This
+	// is used by the signer to restrict the duration an approval
+	// request is valid.
+	uint64 timestamp = 2;
+
+	// The id of the node to pair with.
+	bytes node_id = 3;
```

Comment:
Can we remove this since it looks like it's pulled from the cert on the server side?